plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TransactionLogTail.java

```diff
@@ -44,10 +47,15 @@ public class TransactionLogTail
     private final List<Transaction> entries;
     private final long version;
 
-    private TransactionLogTail(List<Transaction> entries, long version)
+    private final Optional<MetadataEntry> metadataEntry;
```

Comment:
why should this class know about `metadataEntry` & `protocolEntry`?

Reply:
It does not need to. The `getMetadataEntry` and `getProtocolEntry` methods below can be rewritten to get this on the fly from `entries`.

Reply:
@Pluies should we change this to compute metadataEntry and protocolEntry on the fly?

Reply:
Sounds good! On it 👍